### ISSUE TITLE
PNPM upgrades on Windows do not report in WinGet

### ISSUE THREAD

**Issue by josh-hemphill** [pnpm/pnpm#6631]
### Code to reproduce the issue:
```sh
winget install pnpm.pnpm
```
after a new version is available:
```sh
pnpm add -g pnpm
winget upgrade
# Lists pnpm as still being out of date
```

### behavior: 

Now that `winget` is shipped by default in win11 and onwards, it might make sense for pnpm upgrades on windows by default to update their version in winget. 

### Additional information:

 - Windows, macOS, or Linux?: windows

**Comment by zkochan** [comment]
Do you mean the update checker should recommend to update pnpm with winget, if pnpm was installed with winget?

**Comment by josh-hemphill** [comment]
No, Winget stores/pulls installed software versions in the Windows registry, where windows-specific software would usually store their respective settings. So it'd be nice to have some mechanism for pnpm to update its reported version in the windows registry when it updates.

**Comment by Ravel-Ravencroft** [comment]
Just adding a quick update on this... Currently running PNPM v8.13.1 on Windows 10 (OS Build 19045.3996) installed via Winget. Even though the PNPM CLI tells me that there's an update available (Pic 1), Winget doesn't have it listed (Pic 2), even though it's been a few days. If Winget is on a slower Update cadence, What would be the recommended way to use PNPM with a very minimal NodeJS config (i.e.: No NPM or Corepack if possible)?

![image](https://github.com/pnpm/pnpm/assets/54156521/4f64deb0-8b23-4049-9e04-de77d26665e8)

![image](https://github.com/pnpm/pnpm/assets/54156521/78773429-2417-4326-9b41-b73edf3431ab)

**Comment by zkochan** [comment]
The issue with winget now is that we have pnpm v9 now on main branch and we release v8 from v8 branch. But our release script for winget fails for some reason on the v8 branch.

**Comment by Ravel-Ravencroft** [comment]
@zkochan Thanks for the timely reply and info. Don't know what wizardry y'all pulled off, but Winget just let me update pnpm to v8.15.1!

**Comment by sitiom** [comment]
> after a new version is available:
> 
> ```shell
> pnpm add -g pnpm
> winget upgrade
> # Lists pnpm as still being out of date
> ```

Perhaps you should be using the corresponding package manager instead when you upgrade:

```powershell
winget upgrade pnpm
winget list pnpm
# Now shows up-to-date
```

**Comment by zkochan** [comment]
Closing: this is not something pnpm can fix.

WinGet tracks what it installed through the Windows registry, so a pnpm that updates itself cannot report the new version there.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).
